src/Query/Builder.php

```diff
@@ -1805,7 +1807,7 @@ private function aliasIdForQuery(array $values): array
 
         foreach ($values as &$value) {
             if (is_array($value)) {
-                $value = $this->aliasIdForQuery($value);
+                $value = $this->aliasIdForQuery($value, false);
```

Comment:
Note that this appears to be the only recursive call to `aliasIdForQuery()`. All other calls (e.g. preparing projections, wheres) are invoked at the root level.